### DISCUSSION TITLE
Why not DuckDB an in-memory database?

### DISCUSSION THREAD

**Discussion by neolord0** [duckdb/duckdb#14751] [Q&A]
Hello.

You have a sentence on your site that says:

It is a common misconception that DuckDB is an in-memory database. While DuckDB can work in-memory, it is not an in-memory database. DuckDB can make use of available memory for caching, it also fully supports disk-based persistence and offloading larger-than-memory operations to disk.

Could you please provide further explanation on this?

What do you think are the criteria for an in-memory database, and where does DuckDB fall short of these criteria?

**Comment by Tishj** [comment]
I think that text just means that DuckDB is not *just* an in-memory database, that term is too dismissive for what capabilities the system has. That doesn't mean it can't be used as a purely in-memory database.
